### PR TITLE
Devadv 813 xamarin forms upgrade

### PR DESCRIPTION
**Pull request by biozal**
this PR is an update to the latest version of Xamarin Forms and fixes for Xamarin Android:

-Updated Core, iOS, Android, and UWP to Xamarin Forms 5.0
-Fixed Android CSPROJ errors
-Upgraded Android to AndroidX
-Change versions of iOS, Android, and UWP to latest supported versions.
-Ran tests against each platform to make sure they still work.
-Removed open source dialog framework and coded to use built in Xamarin Forms Dialog framework.
-Fixes issues in iOS with Transport Security errors
-Fixes Android HTTP package errors with Dependabot.

### CODE REVIEW

**Comment by rajagp** [comment]
@biozal  Can you also update prerequisites specified in https://github.com/couchbaselabs/mobile-travel-sample/blob/master/content/modules/mobile-travel-sample/pages/csharp/installation/travel-mobile-app.adoc as appropriate

**Comment by biozal** [comment]
> @biozal Can you also update prerequisites specified in https://github.com/couchbaselabs/mobile-travel-sample/blob/master/content/modules/mobile-travel-sample/pages/csharp/installation/travel-mobile-app.adoc as appropriate

I updated the images.  I saw nothing in the document that stated what version of things you should use - only the screen shots that show the package versions - so I updated the two screen shots.

**Comment by rajagp** [comment]
Thanks @biozal - the new screenshots should help but I was referring to the Prerequisites section which lists VS 15.6(?)- Is that even the right version of VS (community edition)? Also,  perhaps we can use the opportunity to list any other pre-requisites that you think would be helpful (for instance, version of Xcode to run a compatible version of iOS simulator)

**Comment by biozal** [comment]
> Thanks @biozal - the new screenshots should help but I was referring to the Prerequisites section which lists VS 15.6(?)- Is that even the right version of VS (community edition)? Also, perhaps we can use the opportunity to list any other pre-requisites that you think would be helpful (for instance, version of Xcode to run a compatible version of iOS simulator)

Added version information and fully break down of Visual Studio, iOS, Android, Mac, Windows, and Hot Reloaded information.

File: content/modules/mobile-travel-sample/pages/csharp/installation/travel-mobile-app.adoc [hunk 1]
```diff
@@ -3,10 +3,25 @@
...
+=== Windows Users
+If you are developing on Windows, we recommend that you use a Windows 10 machine with version 1809 or higher.  Also, note that if you choose Manual or docker installation mode, you should also have *administrative privileges on the Windows box* so you can authorize the installation and running of the required executables.
+
+=== Mac Users with M1 Processor
+Visual Studio and Couchbase are supported on M1 processors via the Rosetta v2 translation layer.  
+
+=== iOS 
+For iOS development, we recommend the latest version of XCode 11 or higher.  
+
+=== Android
```
Comment (rajagp) [inline content/modules/mobile-travel-sample/pages/csharp/installation/travel-mobile-app.adoc]: Android Studio version? 4+?

Comment (biozal) [reply content/modules/mobile-travel-sample/pages/csharp/installation/travel-mobile-app.adoc]: added documentation for that and changed wording to hopefully make it more clear with links to Google's documentation.

**Review by rajagp — APPROVED** [review]

**Pull request merged**